distribution/com.modrinth.ModrinthApp.yml

```diff
@@ -0,0 +1,88 @@
+id: com.modrinth.ModrinthApp
+runtime: org.gnome.Platform
+runtime-version: "42"
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+  - org.freedesktop.Sdk.Extension.node18
+
+command: modrinth-app
+finish-args: 
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create # allows for discord rpc
+  - --device=all
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --socket=x11
+  - --socket=wayland
+
+build-options:
+  append-path: "/usr/lib/sdk/rust-stable/bin:/usr/lib/sdk/node18/bin"
+
+  build-args:
+    - --share=network
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+
+  # for some controller mods (see https://github.com/isXander/Controlify/issues/31)
+  - shared-modules/libusb/libusb.json
+
+  - name: flite
+    buildsystem: autotools
+    config-opts:
+      - --enable-shared
+      - --with-audio=pulseaudio
+    sources:
+      - type: archive
+        url: https://github.com/festvox/flite/archive/refs/tags/v2.2.tar.gz
+        sha512: 1ca2f4145651490ef8405fdb830a3b42e885020a7603d965f6a5581b01bed41047d396b38c2ceab138fc0b28d28078db17acd2b5a84c6444cb99d65c581afa72
+    cleanup:
+      - '/lib/libflite*.a'
+
+  - name: modrinth-app
+    buildsystem: simple
+    build-options:
+      append-path: /run/build/modrinth-app/pnpm
+    build-commands:
+      - pnpm config set store-dir .pnpm-store
+      - cd theseus/theseus_gui && pnpm install
+      - cd theseus/theseus_gui && pnpm build
+      - cd theseus && cargo --offline fetch --manifest-path Cargo.toml
+      - cd theseus && cargo --offline build --release
+      
+      - install -Dm755 theseus/target/release/theseus_gui /app/bin/modrinth-app
+
+      - mkdir -p /app/share/{icons/hicolor/256x256/apps,applications,metainfo}
+
+      - cp theseus/theseus_gui/src-tauri/icons/Square284x284Logo.png /app/share/icons/hicolor/256x256/apps/com.modrinth.ModrinthApp.png 
+      - install -Dm644 com.modrinth.ModrinthApp.metainfo.xml /app/share/metainfo/com.modrinth.ModrinthApp.metainfo.xml
+      - install -Dm644 com.modrinth.ModrinthApp.desktop /app/share/applications/com.modrinth.ModrinthApp.desktop
+
+    sources:
+      - cargo-theseus-sources.json
+
+      - type: shell
+        commands:
+          - mkdir .cargo
+          - cp cargo/config .cargo/
+
+      - type: archive
+        url: https://registry.npmjs.org/@pnpm/linux-x64/-/linux-x64-7.33.5.tgz
+        sha512: d594a9e505b33a5a7b8684645ca762a2c24c6b3e1e477afe8f412d0dd5c8b1b9b9e636e3a19ae0ceeed603d8507f84b081ed12b283ac543bc4bf26d591cab4c5
+        dest: pnpm
```

Comment:
Not building from source here?

Reply:
You need pnpm to build the pnpm, there's no  (easy) way to build it from source

Reply:
Are there aarch64 binaries available upstream? Aarch64 would be a nice-to-have, especially since Asahi Linux is getting pretty damn close to being usable.

Reply:
Yes, there is `https://registry.npmjs.org/@pnpm/linux-arm64/-/linux-arm64-7.33.5.tgz`, that's something that needs to be added. Although I am not sure how the app itself and the minecraft work on arm

Reply:
Actually, there's also `https://registry.npmjs.org/pnpm/-/pnpm-8.6.12.tgz`, for JavaScript version which is architecture independent. The binaries are needed, because they bundle the Node.js, but we already have it from the SDK extension

Reply:
> Are there aarch64 binaries available upstream? Aarch64 would be a nice-to-have

this was a good point, i completely forgot about arm lol. the architecture independent version blt-r mentioned is now being used